pkg/cluster/adminupdate_test.go

```diff
@@ -82,7 +82,9 @@ func TestAdminUpdateSteps(t *testing.T) {
 				"[Action fixMCSCert-fm]",
 				"[Action fixMCSUserData-fm]",
 				"[Action ensureGatewayUpgrade-fm]",
+				"[Condition aroDeploymentReady-fm, timeout 5m0s]",
 				"[Action rotateACRTokenPassword-fm]",
+				"[Action validateACRToken-fm]",
```

Comment:
Since `rotateACRTokenPassword-fm` just updates the secret in the operator namespace and doesn't wait for the operator to add that secret to the global pull secret, we could hit a race condition here if we add a new PUCM step `validateACRToken-fm`, since our operator might not be finished reconciling yet.

I also think that `rotateACRTokenPassword` should confirm that the operator is running and ready before doing anything, since we depend on it.

It might make sense to have it all in the same PUCM task and wait for the operator to reconcile, then validate before returning. It might also make sense to have this validation built in to the operator instead, since we use it to update the cluster pull secret. What are your thoughts?

Reply:
I was hoping to cover the "operator is running and ready" by using the Condition aroDeploymentReady rather than adding a wait-loop to rotateACRTokenPassword, since Conditions already have the ability to timeout. What if I made validateACRToken also a Condition and have it wait for the operator to finish reconciliation?

Reply:
I think that would work!